static/app/components/events/errorItem.tsx

```diff
@@ -69,25 +70,38 @@ export function ErrorItem({error}: ErrorItemProps) {
       );
     }
 
-    return Object.entries(data).map(([key, value]) => ({
-      key,
-      value,
-      subject: keyMapping[key] || startCase(key),
-      meta: getMeta(data, key),
-    }));
-  }, [error.data]);
+    return Object.entries(data)
+      .map(([key, value]) => ({
+        key,
+        value,
+        subject: keyMapping[key] || startCase(key),
+        meta: key === 'image_name' ? meta?.image_path?.[''] : meta?.[key]?.[''],
+      }))
+      .filter(d => {
+        if (!d.value && !!d.meta) {
+          return true;
+        }
+        return !!d.value;
+      });
+  }, [error.data, meta]);
 
   return (
     <StyledListItem>
       <OverallInfo>
         <div>
-          {!error.data?.name || typeof error.data?.name !== 'string' ? null : (
+          {meta?.data?.name?.[''] ? (
+            <AnnotatedText value={error.message} meta={meta?.data?.name?.['']} />
+          ) : !error.data?.name || typeof error.data?.name !== 'string' ? null : (
             <Fragment>
               <strong>{error.data?.name}</strong>
               {': '}
             </Fragment>
           )}
-          {error.message}
+          {meta?.message?.[''] ? (
+            <AnnotatedText value={error.message} meta={meta?.message?.['']} />
+          ) : (
+            error.message
+          )}
```

Comment:
not a blocker: How about we make the AnnotatedText fallback to render plain value if the meta is not there so that we don't have to do these checks all over the place?

Reply:
it is not that simple, because in same cases we pass a react element as a value but we can think about something better in the suture

Reply:
why can't we render that react element?

Reply:
a react element is always an object and this doesn't play well with the logic we currently have to display the `ValueElement` . That's the reason